### ISSUE TITLE
[Platform Request] MSX Computers

### ISSUE THREAD

**Issue by Redlolz** [hsoft/collapseos#23]
Maybe support for compiling to MSX computers could be added? since they're z80 computers and already have all the stuff you need. I could test some things since I have access to msx 1 and msx 2 computers, although I don't know how to compile for them.

MSX 1 Specs:
- Z80A @ 3.58 MHz
- RAM minimum of 8kb, most computers use 32kb or 64kb
- TMS9918 family of Video Display Processors
- AY-3-8910 Sound Chip

ps the msx 2 is backwards compatible with the first version

**Comment by hsoft** [comment]
Yes, recipes for compatible machines is the kind of stuff we want to add to Collapse OS. At the moment, things are a bit chaotic because many people begin working on many hardware at once. If we want to avoid code duplication, we might want to go a bit slow. It looks like someone will come up with ZX Spectrum support rather soon. Maybe see what it's going to look like first.

That being said, if you want to start right now, go ahead, but be aware that we might have to adapt your code to code that isn't in the repo yet. There's going to be "PR race conditions".

**Comment by insensitiveclod** [comment]
Second on the MSX platform. I have a few of these laying around as well and would love to see a port.
I havent done any low-level MSX coding when I grew up on the machines, but I do have a ton of technical docs that I saved/collected over the years. Might give some of this a go. 

CHeers

**Comment by Redlolz** [comment]
I looked at what was needed for this port and it looks like only the VDP and the keyboard part of the Sega Master System needs to be rewritten and then we should have a working port. Although I'm unsure if what we need is a bios replacement or a cartridge that boots on the MSX platform.
